File: app/src/main/java/com/kelsos/mbrc/repository/ConnectionRepository.java
package com.kelsos.mbrc.repository;

import com.kelsos.mbrc.data.ConnectionSettings;

import java.util.List;

public interface ConnectionRepository {
  void save(ConnectionSettings settings);

  void delete(ConnectionSettings settings);

  void update(ConnectionSettings settings);

  ConnectionSettings getDefault();

  void setDefault(ConnectionSettings settings);

  List<ConnectionSettings> getAll();

  long count();

  long getDefaultId();
}

File: app/src/main/kotlin/com/kelsos/mbrc/repository/ConnectionRepository.kt
package com.kelsos.mbrc.repository

import com.kelsos.mbrc.data.ConnectionSettings

interface ConnectionRepository {
  suspend fun save(settings: ConnectionSettings)

  suspend fun delete(settings: ConnectionSettings)

  suspend fun update(settings: ConnectionSettings): Boolean

  suspend fun getAll(): List<ConnectionSettings>

  suspend fun count(): Long

  suspend fun setDefault(settings: ConnectionSettings)

  suspend fun getDefault(): ConnectionSettings?

  val defaultId: Long
}
